keystoremanager/import_export.go

```diff
@@ -75,43 +75,27 @@ func (km *KeyStoreManager) ImportRootKey(source io.Reader, keyID string) error {
 		return err
 	}
 
-	if err = km.rootKeyStore.Add(keyID, pemBytes); err != nil {
+	if err = km.rootKeyStore.Add(keyID+"_root", pemBytes); err != nil {
 		return err
 	}
 
 	return err
 }
 
-func moveKeysWithNewPassphrase(oldKeyStore, newKeyStore *trustmanager.KeyFileStore, outputPassphrase string) error {
+func moveKeys(oldKeyStore, newKeyStore *trustmanager.KeyFileStore) error {
 	// List all files but no symlinks
-	for _, f := range oldKeyStore.ListFiles(false) {
-		fullKeyPath := strings.TrimSpace(strings.TrimSuffix(f, filepath.Ext(f)))
-		relKeyPath := strings.TrimPrefix(fullKeyPath, oldKeyStore.BaseDir())
-		relKeyPath = strings.TrimPrefix(relKeyPath, string(filepath.Separator))
-
-		pemBytes, err := oldKeyStore.Get(relKeyPath)
+	for _, f := range oldKeyStore.ListKeys() {
+		pemBytes, err := oldKeyStore.GetKey(f)
 		if err != nil {
 			return err
 		}
 
-		block, _ := pem.Decode(pemBytes)
-		if block == nil {
-			return ErrNoValidPrivateKey
+		alias, err := oldKeyStore.GetKeyAlias(f)
+		if err != nil {
+			return err
 		}
 
-		if !x509.IsEncryptedPEMBlock(block) {
-			// Key is not encrypted. Parse it, and add it
-			// to the temporary store as an encrypted key.
-			privKey, err := trustmanager.ParsePEMPrivateKey(pemBytes, "")
-			if err != nil {
-				return err
-			}
-			err = newKeyStore.AddEncryptedKey(relKeyPath, privKey, outputPassphrase)
-		} else {
-			// Encrypted key - pass it through without
-			// decrypting
-			err = newKeyStore.Add(relKeyPath, pemBytes)
-		}
+		err = newKeyStore.AddKey(f, alias, pemBytes)
```

Comment:
how do you know if the current key is encrypted or not, and if you have to encrypt it before storing?